src/boosting/gbdt.cpp

```diff
@@ -377,9 +298,23 @@ bool GBDT::TrainOneIter(const score_t* gradients, const score_t* hessians) {
     Boosting();
     gradients = gradients_.data();
     hessians = hessians_.data();
+  } else if (gradients != nullptr) {
+    // use customized objective function
+    CHECK(hessians != nullptr && objective_function_ == nullptr);
+    // and will be only used for GOSS (No? But copying gradients will not have effect when bagging)
+    // CHECK(config_->boosting==std::string("goss") || config_->data_sample_strategy==std::string("goss"));
```

Comment:
>boosting==std::string("goss") || config_->data_sample_strategy==std::string("goss")); [](http://example.com/codeflow?start=22&length=85)

remove the unused code instead of commenting them

Reply:
Thanks! Already removed.